tests/eigensolve_test_gtest.hpp

```diff
@@ -0,0 +1,89 @@
+#include <gtest/gtest.h>
+
+using test_t = ::testing::tuple<QudaEigType, QudaBoolean, QudaBoolean, QudaBoolean, QudaEigSpectrumType>;
+
+class EigensolveTest : public ::testing::TestWithParam<test_t>
+{
+protected:
+  test_t param;
+
+public:
+  EigensolveTest() : param(GetParam()) { }
+};
+
+bool is_chiral(QudaDslashType type)
+{
+  switch (type) {
+  case QUDA_DOMAIN_WALL_DSLASH:
+  case QUDA_DOMAIN_WALL_4D_DSLASH:
+  case QUDA_MOBIUS_DWF_DSLASH:
+  case QUDA_MOBIUS_DWF_EOFA_DSLASH: return true;
+  default: return false;
+  }
+}
+
+bool skip_test(test_t param)
+{
+  // dwf-style solves must use a normal solver
+  if (is_chiral(dslash_type) && (::testing::get<1>(param) == QUDA_BOOLEAN_FALSE)) return true;
+  return false;
+}
+
+std::vector<double> eigensolve(test_t test_param);
+
+TEST_P(EigensolveTest, verify)
+{
+  if (skip_test(GetParam())) GTEST_SKIP();
+  auto tol = 5 * eig_param.tol;
```

Comment:
What's this factor of 5 for?

Reply:
The IRAM eigensolver will sometimes report convergence with tolerances slightly higher that requested. The same phenomenon occurs in ARPACK. This fudge factor prevents failure when IRAM has solved to say 2e-6 when 1e-6 is requested. The solution to avoid this is to use a Krylov space (n-kr) about 3-4 times the size of the search space (n-ev), or use a well chosen Chebyshev polynomial, or use a tighter than necessary tolerance.

Reply:
Ok, could you leave a comment to this effect then and only apply the fudge factor if doing IRAM?

Reply:
Done 9b58a9770